docs/syscall_descriptions.md

```diff
@@ -127,6 +127,65 @@ change in descriptions for a particular syscall, the programs that are already i
 the corpus will be kept there, unless you manually clear them out (for example by
 removing the `corpus.db` file).
 
+<div id="tips"/>
+
+## Description tips and FAQ
+
+<div id="names"/>
+
+### Syscall, struct, field, flags names
+
+Stick with existing kernel names for things, don't invent new names if possible.
+
+For example, if there is an existing enum `v4l2_buf_type` in the kernel headers,
+use this name for flags in descriptions as well. The same for structs, unions,
+fields, etc. For syscall descriminations, use the discriminating command name
```

Comment:
discrimination

Reply:
@melver we need somebody who is more native speaker than we are :)
I mean discriminating as a participle for the command noun. Is it any English at all?

Reply:
I meant that we should spell `d_i_scrimi...` here, but I don't know if it's the right word in general :)

Reply:
"syscall discriminations" and "discriminating command"  sort of make sense, if we know the terminology. But it's not really intuitive. "discriminating" isn't documented much elsewhere in the codebase.

For the use in syzkaller, I don't actually know if discriminating is any meaningful word.

What is the actual meaning of  '$xxx'? Is it just a tag? If so, discriminating is confusing, unless it's an already established terminology, but I can't find evidence of that. Why "discriminating"?

I've encountered "discriminating" in the context of "discriminating unions" (https://en.wikipedia.org/wiki/Tagged_union), although find "tagged" union or "variant" nicer because it's shorter. "Discriminating" has too many characters and is easy to misspell. ;-)

Would "syscall tags" and "tagged command" be more intuitive?

Reply:
Done for s/descriminations/discriminations/

Re more general naming question. The term we need is "what is cmd for fcntl".
If you look at fcntl:
https://man7.org/linux/man-pages/man2/fcntl.2.html
It's really a dozen of different syscalls with different functions, arguments and return values (don't ask me why they were shoved into a single syscall). The actual syscall is determined by the value of cmd argument. Now if we generalize this to ioctl/setsockopt/etc, what is the collective name for such syscalls and for such "cmd" (as opposed to other arguments that are just arguments).

And we also use such syscall discriminations for another purpose -- which is just having several sets of different arguments (specializations) for some reason, but there is not necessary such "cmd" argument and not necessary different actual syscalls. I think this meaning is closer to what you called "variant" (just variants with no particular meaning beyond that).
But for this paragraph in the doc we still need the term that refers to "what is cmd for fcntl".

Reply:
> "Discriminating" has too many characters and is easy to misspell. ;-)

Indeed. Unfortunately linter did not catch my misspelling :)

```
syzkaller$ find -name "*.go" -exec grep "discrimi" {} \; -print
	"discriminatoire", "discriminate",
	"discriminatorie", "discriminate",
	"discrimanatory", "discriminatory",
	"discriminacion", "discrimination",
	"discriminitory", "discriminatory",
	"diskrimination", "discrimination",
	"discrimianted", "discriminated",
	"discriminante", "discriminate",
	"discriminatie", "discriminate",
	"discriminatin", "discrimination",
	"descriminant", "discriminant",
	"discriminare", "discriminate",
	"discriminted", "discriminated",
	"discrimante", "discriminate",
	"discrimiate", "discriminate",
	"discrimine", "discriminate",
./vendor/github.com/golangci/misspell/words.go
```

Reply:
That's tricky. My guess is that "syscall variant" might actually be more intuitive then.

Let's see how any of these sound:

- "syscall discrimination" (what you have now, but not at all intuitive because "discrimination" also has other overloaded meanings in English which are not at all relevant: https://dictionary.cambridge.org/dictionary/english/discrimination)

- "syscall variant" (similar, but the word "variant" is clearer and its meaning is not as overloaded)

- "syscall command" (straightforward, but does it cover all your uses?)

- "syscall operation" (similar to "command")

- "syscall case" (maybe closer to what you might see in code)

- "syscall specialization" (close to OOP terminology of specialization)

All I would suggest is to avoid "discrimination" if you can, and pick something more intuitive.

Reply:
Let's do "variant". PTAL.